src/proxy/outbound.rs

```diff
@@ -357,16 +434,69 @@ impl OutboundConnection {
             }
             debug!("built request as passthrough; no upstream found");
             return Ok(Request {
-                protocol: Protocol::TCP,
+                protocol: OutboundProtocol::TCP,
                 source: source_workload,
                 hbone_target_destination: None,
                 actual_destination_workload: None,
                 intended_destination_service: None,
                 actual_destination: target,
                 upstream_sans: vec![],
+                final_sans: vec![],
             });
         };
 
+        // Check whether we are using an E/W gateway and sending cross network traffic
+        if let Some(ew_gtw) = &us.workload.network_gateway {
+            if us.workload.network != source_workload.network {
+                let svc = us
+                    .destination_service
+                    .as_ref()
+                    .expect("Workloads with network gateways must be service addressed.");
+
+                let (actual_destination, upstream_sans, final_sans) = match &ew_gtw.destination {
+                    Destination::Address(address) => (
+                        SocketAddr::from((address.address, ew_gtw.hbone_mtls_port)),
+                        us.workload_and_services_san(),
+                        us.service_sans(),
+                    ),
+                    Destination::Hostname(host) => {
+                        let us_gtw = self
+                            .pi
+                            .state
+                            .fetch_upstream_by_host(
```

Comment:
If we follow the above comment about following fetch_watchpoint this would probably end up calling `find_hostname --> find_upstream_from_service`